tests/unit/python/foglamp/services/core/api/test_support.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+
+import json
+import pathlib
+
+from unittest.mock import patch
+
+from aiohttp import web
+import pytest
+
+from foglamp.services.core import routes
+from foglamp.services.core.api import support
+
+__author__ = "Ashish Jabble"
+__copyright__ = "Copyright (c) 2018 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+@pytest.allure.feature("unit")
+@pytest.allure.story("api", "bundle-support")
+class TestBundleSupport:
+
+    @pytest.fixture
+    def client(self, loop, test_client):
+        app = web.Application(loop=loop)
+        # fill the routes table
+        routes.setup(app)
+        return loop.run_until_complete(test_client(app))
+
+    @pytest.fixture
+    def support_bundles_dir_path(self):
+        return pathlib.Path(__file__).parent
+
+    @pytest.mark.parametrize("data, expected_content, expected_count", [
+        (['support-180301-13:35:23.tar.gz', 'support-180301-13:25:23.tar.gz'], {'bundles': ['support-180301-13:35:23.tar.gz', 'support-180301-13:25:23.tar.gz']}, 2),
+        (['support-180301-13:35:23.tar.gz', 'foglamp.txt'], {'bundles': ['support-180301-13:35:23.tar.gz']}, 1),
+        (['foglamp.txt'], {'bundles': []}, 0),
+        ([], {'bundles': []}, 0)
+    ])
+    async def test_get_support_bundle(self, client, support_bundles_dir_path, data, expected_content, expected_count):
+        path = support_bundles_dir_path / 'support'
+        with patch.object(support, '_get_support_dir', return_value=path):
+            with patch('os.walk') as mockwalk:
+                mockwalk.return_value = [(path, [], data)]
+                resp = await client.get('/foglamp/support')
+                assert 200 == resp.status
+                res = await resp.text()
+                jdict = json.loads(res)
+                assert expected_count == len(jdict['bundles'])
+                assert expected_content == jdict
+            mockwalk.assert_called_once_with(path)
+
+    # FIXME: issue with URL having colon in it even after encoded with %3A manually or urllib parser
+    async def test_get_support_bundle_by_name(self, client, support_bundles_dir_path):
```

Comment:
https://github.com/aio-libs/aiohttp/issues/2670
https://github.com/aio-libs/aiohttp/commit/06ed15ecb2a4700a08f4f2ff4b03bd751ca979f6 /cc @ashish-jabble

Reply:
@MarkRiddoch As per specs mentioned in FOGL-1126 (The file will be called support-<YYMMDD>-<HH:MM:SS>.tar.gz).So I had added tests and example with this naming convention but somehow there is aiohttp bug on `v2.3.8` and bug has fixed in `2.3.9` **BUT with yarl dep 1.0.0**

So if we need to follow the same file naming convention,  we need to fix below things
1) aiohttp dep set to 2.3.9
2) and yarl dep explicitly set to 1.0.0

Please let me know, should I need to do above steps to make it work? or simplest and easiest way to just change the file naming convention we had in specs and do it like "support-YYMMDD-HH-MM-SS" or "support_YYMMDD_HH_MM_SS" or anything.

Reply:
I'm not too worried about the exact naming, so rather than force this update let's go for one of the alternate naming conventions. I would suggest support-YYMMDD-HH-MM-SS.